src/boot_O2_g3/fault_drawer.c

```diff
@@ -1,28 +1,64 @@
+/**
+ * @file fault_drawer.c
+ *
+ * Implements routines for drawing text with a fixed font directly to a framebuffer, used in displaying
+ * the crash screen implemented by fault.c
+ */
+
+#include "fault.h"
+#include "fault_internal.h"
 #include "global.h"
 #include "vt.h"
 
+typedef struct {
+    /* 0x00 */ u16* fb;
+    /* 0x04 */ u16 w;
+    /* 0x06 */ u16 h;
+    /* 0x08 */ u16 yStart;
+    /* 0x0A */ u16 yEnd;
+    /* 0x0C */ u16 xStart;
+    /* 0x0E */ u16 xEnd;
+    /* 0x10 */ u16 foreColor;
+    /* 0x12 */ u16 backColor;
+    /* 0x14 */ u16 cursorX;
+    /* 0x16 */ u16 cursorY;
+    /* 0x18 */ const u32* fontData;
+    /* 0x1C */ u8 charW;
+    /* 0x1D */ u8 charH;
+    /* 0x1E */ s8 charWPad;
+    /* 0x1F */ s8 charHPad;
+    /* 0x20 */ u16 printColors[10];
+    /* 0x34 */ u8 escCode; // bool
+    /* 0x35 */ u8 osSyncPrintfEnabled;
+    /* 0x38 */ FaultDrawerCallback inputCallback;
+} FaultDrawer; // size = 0x3C
+
 extern const u32 sFaultDrawerFont[];
 
-FaultDrawer sFaultDrawerStruct;
+FaultDrawer sFaultDrawer;
+
+FaultDrawer* sFaultDrawerRef = &sFaultDrawer;
```

Comment:
```suggestion
FaultDrawer sFaultDrawerInstance;

FaultDrawer* sFaultDrawer = &sFaultDrawerInstance;
```
Mirrors what happens with `faultMgr` which I think is better, though it is worth noting that this does differ from `OoT` since they just use the instance directly, and don't have the reference variable.

Reply:
Oh I actually swapped these from what `Fault` does. The `Instance` suffix should be on the pointer.